src/ProductConstructionService/ProductConstructionService.Api/Queue/WorkItems/WorkItem.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+
+namespace ProductConstructionService.Api.Queue.WorkItems;
+
+[JsonDerivedType(typeof(TextWorkitem), typeDiscriminator: nameof(TextWorkitem))]
+public abstract class WorkItem
+{
+    public Guid Id { get; } = Guid.NewGuid();
```

Comment:
I don't follow here - shouldn't we be able to deserialize the Id from the queue message too?

Reply:
I am fine with using our custom ID but I more meant that this class looks like it won't ever deserialize the ID even if we have it in the queue - it will just get a new GUID instead?

Reply:
ah right, that's a fair point, my bad